aopy/analysis/base.py

```diff
@@ -1667,3 +1667,46 @@ def calc_corr2_map(data1, data2, knlsz=15, align_maps=False):
     NCC[nan_idx1] = np.nan
     
     return NCC, shifts
+
+
+'''
+Statistics 
+'''
+def permutation_test(sample1, sample2, statistic, num_permutations):
+    """
+    Conducts a permutation test to compute the p-value for the t-test statistic.
+
+    Parameters:
+        sample1 (array-like): distribution 1.
+        sample2 (array-like): distribution 2.
+        statistic (function): The function to compute the test statistic.
+                              This function should take two arrays as input
+                              (representing the samples) and return the test statistic.
+        num_permutations (int): The number of permutations to perform.
+
+    Returns:
+        p_value (float): The p-value for the permutation test.
```

Comment:
can you be more descriptive about the p-value, is it the probability of the null distribution being above the original sample statistic?